src/com/palmergames/bukkit/towny/command/NationCommand.java

```diff
@@ -1847,4 +1859,211 @@ public void nationRename(Player player, Nation nation, String newName) {
 	}
 
 
+    /**
+     * Wrapper for the nationSpawn() method. All calls should be through here
+     * unless bypassing for admins.
+     *
+     * @param player
+     * @param split
+     * @throws TownyException
+     */
+    public static void nationSpawn(Player player, String[] split) throws TownyException {
+
+        try {
+
+            Resident resident = TownyUniverse.getDataSource().getResident(player.getName());
+            Nation nation;
+            String notAffordMSG;
+
+            // Set target nation and affiliated messages.
+            if (split.length == 0) {
+
+                if (!resident.hasTown()) {
+                    TownyMessaging.sendErrorMsg(player, TownySettings.getLangString("msg_err_dont_belong_nation"));
+                    return;
+                }
+
+                if(!resident.getTown().hasNation()){
+					TownyMessaging.sendErrorMsg(player, TownySettings.getLangString("msg_err_dont_belong_nation"));
+					return;
+				}
+
+                nation = resident.getTown().getNation();
+                notAffordMSG = TownySettings.getLangString("msg_err_cant_afford_tp");
+
+                nationSpawn(player, split, nation, notAffordMSG);
+
+            } else {

```

Comment:
I'm not certain that this is something needed. This section of town spawning is for public spawning, which I haven't seen yet if you've set up public/non-public settings for nations. I don't think that the intent of the tickets requesting this was for spawning to other players' nations, just to a player's own nation.